src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -287,7 +287,7 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
                             });
                     }
 
-                    if (IfFunction.CanCheckIfReturn(func))
+                    if (RuntimeHelpers.CanAssertReturn(func))

```

Comment:
note this will go away with #1113

Reply:
Then it might be a good idea until #1113 is merged. I don't see this change working without this skip.

Reply:
Thats a good suggestion but I don't think it fits for this situation, unless we "unwrap" the table value after EvalVisitor.
Expression: `[{a:1}, {a:true, b:2}]`
The binder produces a `*[a:n]` table type, which should be `*[a:n, b:n]`. I can wrap the result back to `*[a:n]` but then we we cant call `recordvalue.b`.
If I am missing something, please let me know.

Reply:
I am wondering why we don't compute the correct type in the binder. I think it has all the information to compute it. Shouldn't we? @MikeStall